frontend/utils/translation.ts

```diff
@@ -13,23 +16,37 @@ export function formatMultiPartSpan (t: ComposerTranslation, key: string, classe
   return `<span>${parts.join(' ')}</span>`
 }
 
-// Translation with default. Needed if we want to default to '' as the translation lib does not support to default to ''
+/** Translation with default. Needed if we want to default to '' as the translation lib does not support to default to '' */
 export function tD (t: ComposerTranslation, path : string, options?: any, d:string = '') : string {
-  const translation = t(path, 'NOT_FOUND', { ...options, missingWarn: false })
-  return (translation === 'NOT_FOUND') ? d : translation
+  if (typeof options === 'number') {
+    options = { plural: options }
+  }
+  const translation = t(path, NOT_FOUND, { ...options, missingWarn: false })
+  return (translation === NOT_FOUND) ? d : translation
 }
 
-// get a translation at a specific index, if it exists
+function tOfWithEOA (t: ComposerTranslation, path : string, index : number, options?: any) : string {
+  return tD(t, `${path}[${index}]`, options, END_OF_ARRAY)
+}
+
+/** gets a translation at a specific index, if it exists */
 export function tOf (t: ComposerTranslation, path : string, index : number, options?: any) : string {
-  return tD(t, `${path}[${index}]`, options)
+  const element = tOfWithEOA(t, path, index, options)
+  return (element === END_OF_ARRAY) ? '' : element
 }
 
-// get all translations for an array
+/** returns an array of translations (with only one element if the translation data is a simple string) */
 export function tAll (t: ComposerTranslation, path : string, options?: any) : string[] {
   const list: string[] = []
-  let index = 0
-  for (let value = tOf(t, path, index); value; value = tOf(t, path, ++index, options)) {
-    list.push(value)
+  if (tD(t, path, options, NOT_FOUND) !== NOT_FOUND) {
+    // the data is a string, we push the translation
+    list.push(tD(t, path, options))
+  } else {
+    // the data is an array
+    let index = 0
+    for (let value = tOfWithEOA(t, path, index, options); value !== END_OF_ARRAY; value = tOfWithEOA(t, path, ++index, options)) {
```

Comment:
I am pretty sure the previous version without the END_OF_ARRAY did also work

Reply:
The problem is that `''` caused `tAll()` to stop when a translation part was empty (which might happen for example because in some language this part is at a different place (like a verb in German that moves to the end)).
So I made this change to make sure that `tAll()` really stops at the end of the array and never before.

Reply:
I don't see the need for empty strings in the translations. 
Example: 
English would have the bold text in the middle
["This is my ", "bold, "test"]
If in another language the bold text is at the end then it would only have two text parts
["Am Ende ist der Text", "fett"]

... I don't see a use case where we would have an empty string in the middle

Reply:
> I don't see a use case where we would have an empty string in the middle

Consider this two sentences with bold and links:
- As we like both, the [chocolate](chocolink) that you bring _**may**_ be dark or with milk.
- Da wir beide mögen, _**darf**_ die [Schokolade](chocolink), die du bringst, dunkel oder mit Milch sein.

Their arrays need empty elements:

```TS
["As we like both, the", "chocolate", "that you bring", "may", "be dark or with milk."]
```
```TS
["Da wir beide mögen", "", "", "darf", "die", "Schokolade", "die du bringst, dunkel oder mit Milch sein"]
```
And even if we don't talk about chocolate on beaconcha.in, I am convinced that stopping the scan of the array at the first empty element was a bug.